application/src/main/kotlin/net/corda/v5/application/crypto/SigningService.kt

```diff
@@ -20,10 +21,36 @@ import java.util.*
 @DoNotImplement
 interface SigningService {
 
+    /**
+     * Using the provided signing [PublicKey], internally chooses the most appropriate signing spec,
+     * looks up the matching [PrivateKey] and signs the data, selecting the most appropriate signing type.
+     * This overload is preferred; see the other overload if you need to control the signing type.
+     *
+     * @param payload The data to sign over using the chosen key.
+     * @param metadata Optional map of string configuration metadata to be included in the signature. If the metadata
+     *                 is null, or if there are no know entries in the map, that's equivalent to calling `sign` without
+     *                 metadata.
+     * @param publicKey The [PublicKey] partner to an internally held [PrivateKey], either derived from the node's
+     * primary identity, or previously generated via the freshKey method. If the [PublicKey] is actually
+     * a [CompositeKey], the first leaf signing key hosted by the node is used.
```

Comment:
This could be expensive to do, but it does make sense from a usability perspective.